src/units/abilities.cpp

```diff
@@ -429,19 +411,44 @@ unit::recursion_guard& unit::recursion_guard::operator=(unit::recursion_guard&&
 unit::recursion_guard::~recursion_guard()
 {
 	if(parent) {
-		assert(parent->num_recursion_ > 0);
-		parent->num_recursion_--;
+		assert(!parent->open_queries_.empty());
+		parent->open_queries_.pop_back();
 	}
 }
 
+bool unit::get_self_ability_bool(const config& ability, const std::string& tag_name, const map_location& loc) const
```

Comment:
Please keep these where they were in the file, so that the diff compares the old and new implementations of get_self_ability_bool.

Reply:
done